src/java.base/share/classes/sun/nio/ch/SocketChannelImpl.java

```diff
@@ -403,6 +405,20 @@ private void throwConnectionReset() throws SocketException {
 
     @Override
     public int read(ByteBuffer buf) throws IOException {
+        if (!SocketReadEvent.enabled()) {
+            return implRead(buf);
+        }
+        int nbytes = 0;
+        long start = SocketReadEvent.timestamp();
+        try {
+            nbytes = implRead(buf);
+        } finally {
+            SocketReadEvent.checkForCommit(start, nbytes, getRemoteAddress(), 0);
```

Comment:
This will need to be changed to use remoteAddress(), can't use getRemoteAddress() as it might throw.